packages/block-library/src/navigation/edit/index.js

```diff
@@ -142,25 +144,44 @@ function Navigation( {
 		`navigationMenu/${ ref }`
 	);
 
-	const { innerBlocks, isInnerBlockSelected, hasSubmenus } = useSelect(
+	const {
+		hasUncontrolledInnerBlocks,
+		uncontrolledInnerBlocks,
+		controlledInnerBlocks,
+		isInnerBlockSelected,
+		hasSubmenus,
+	} = useSelect(
 		( select ) => {
-			const { getBlocks, hasSelectedInnerBlock } = select(
+			const { getBlock, getBlocks, hasSelectedInnerBlock } = select(
 				blockEditorStore
 			);
-			const blocks = getBlocks( clientId );
-			const firstSubmenu = !! blocks.find(
-				( block ) => block.name === 'core/navigation-submenu'
-			);
+
+			// This relies on the fact that `getBlock` won't return controlled
+			// inner blocks, while `getBlocks` does. It might be more stable to
+			// introduce a selector like `getUncontrolledInnerBlocks`, just in
+			// case `getBlock` is fixed.
+			const _uncontrolledInnerBlocks = getBlock( clientId ).innerBlocks;
```

Comment:
This doesn't seem to work. In my testing despite the entity being created and the code view of the markup showing no inner blocks, the block editor store wasn't in sync and thus still showed inner blocks.

Easy to check:

```
wp.data.select('core/block-editor').getBlocks()
```

This causes problems later on.